### PR TITLE
Save backtest history in sqlite.

### PR DESCRIPTION
**Pull request by tekwani**
Save backtest history in sqlite.
Get backtest history by strategyId.
Get backtest history strategy details by id.
Set backtest strategy as favorite.

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -45,6 +56,8 @@ class DataServer {
 
     this.port = port
     this.wss = null
+
+    if (sqlitePath) sqliteDb.connectDb(sqlitePath)
```
Comment (tarcisiozf) [inline lib/server.js:60]: I think we should handle possible errors when connecting, and by that also moving this to `open` method instead

File: lib/db/bt_dao.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+const getBtHistory = async (strategyId) => {
+  const query = 'SELECT * FROM bt_history where strategyId=?'
+  const values = [strategyId]
+
+  const btHistory = await sqliteDb.queryData(query, values)
+  if (btHistory.length === 0) {
+    return btHistory
+  }
+  const normilizedBtHistory = btHistory.map((bt) => {
```
Comment (tarcisiozf) [inline lib/db/bt_dao.js:54]: typo in normalized

**Review by tarcisiozf — CHANGES_REQUESTED** [review]

File: lib/db/bt_dao.js (hunk 2 above)
Comment (tekwani) [reply lib/db/bt_dao.js]: fixed.

File: lib/server.js (hunk 1 above)
Comment (tekwani) [reply lib/server.js]: fixed.

**Review by tarcisiozf — APPROVED** [review]

**Pull request merged**